src/Routes/employerRoutes.ts

```diff
@@ -0,0 +1,161 @@
+import { Router, Request, Response } from "express";
+import mongoose, { CallbackError } from "mongoose";
+
+// Middlewarex
+import { verifyToken } from "../Middleware/Authorization";
+
+// Models
+import EmployerModel from "../Models/Employer";
+
+// Instantiate the router
+const employerRoutes = Router();
+
+// Use Middleware
+employerRoutes.use(verifyToken);
+
+/**
+ * Route for getting all companies.
+ * @name GET /
+ * @function
+ * @alias module:Routes/companyRoutes
+ * @property {Request} _ Express Request
+ * @property {Response} res Express Response
+ * @returns {Promise<any>}
+ */
+employerRoutes.get("/", async (_: Request, res: Response): Promise<any> => {
+  try {
+    const companies = await EmployerModel.find({}, { __v: 0 })
+      .populate({ path: "jobs", select: "title type location" })
+      .exec();
+    if (!companies) return res.status(200).send("No employers exist");
+
+    return res.status(200).send(companies);
+  } catch (err) {
+    return res.status(500).send(err);
+  }
+});
+
+/**
+ * Route for getting a company by id.
+ * @name GET /:id
+ * @function
+ * @alias module:Routes/companyRoutes
+ * @property {Request} req Express Request
+ * @property {Response} res Express Response
+ * @returns {Promise<any>}
+ */
+employerRoutes.get(
+  "/:id",
+  async (req: Request, res: Response): Promise<any> => {
+    const { id } = req.params;
+
+    try {
+      const company = await EmployerModel.findOne({ _id: id }, { __v: 0 })
+        .populate({ path: "jobs", select: "title type location" })
+        .exec();
+
+      if (!company)
+        return res.status(200).send("No employer exists with that ID");
+
+      return res.status(200).send(company);
+    } catch (err) {
+      return res.status(500).send(err);
+    }
+  }
+);
+
+/**
+ * Route for creating a company.
```

Comment:
This is something we need to design out more. If we allow employers to make companies it creates a host of workflow issues and employer account creation for companies. Just seems strange, we should design an Admin user. One for the admin page we will eventually make and 2 so they can do company creation for companies wanting to register, or some similar workflow. Let me know thoughts.

Reply:
I agree